### PR TITLE
Add support for other STM32F4 UARTs

### PR DESCRIPTION
**Pull request by marshall**
I finished this before release, but it wasn't ready in time for the cut off.. We should rebase openkosmosorg/kubos-hal-stm32f405rg#1 on this once it's merged

### CODE REVIEW

File: source/uart.c [hunk 1]
```diff
@@ -148,58 +150,101 @@ void kprv_uart_dev_init(KUARTNum uart)
...
 
-void USART6_IRQHandler(void)
+static inline void uart_irq_handler(KUARTNum uart)
 {
-    static portBASE_TYPE task_woken;
-    KUART *k_uart = kprv_uart_get(K_UART6);
+    portBASE_TYPE task_woken = pdFALSE;
+    USART_TypeDef *dev = uart_dev(uart);
+    KUART *k_uart = kprv_uart_get(uart);
+    if (!dev || !uart) {
```
Comment (rplauche) [inline source/uart.c]: Should this be
```
if (!dev || !k_uart) {
```

Comment (marshall) [inline source/uart.c]: yup, good catch

File: source/uart.c [hunk 2]
```diff
@@ -148,58 +150,101 @@ void kprv_uart_dev_init(KUARTNum uart)
...
 
-void USART6_IRQHandler(void)
+static inline void uart_irq_handler(KUARTNum uart)
 {
-    static portBASE_TYPE task_woken;
-    KUART *k_uart = kprv_uart_get(K_UART6);
+    portBASE_TYPE task_woken = pdFALSE;
+    USART_TypeDef *dev = uart_dev(uart);
+    KUART *k_uart = kprv_uart_get(uart);
+    if (!dev || !k_uart) {
```
Comment (rplauche) [inline source/uart.c]: NULL checks should get added to the other functions which call `uart_dev` or `kprv_uart_get`.

**Comment by rplauche** [comment]
This looks good to me. Do we want to wait on #8 before we merge this one?

**Pull request merged**